api/handlers/exporter.go

```diff
@@ -0,0 +1,108 @@
+package handlers
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	spectypes "github.com/ssvlabs/ssv-spec/types"
+
+	"github.com/ssvlabs/ssv/api"
+	exporterapi "github.com/ssvlabs/ssv/exporter/api"
+	"github.com/ssvlabs/ssv/exporter/convert"
+	ibftstorage "github.com/ssvlabs/ssv/ibft/storage"
+	"github.com/ssvlabs/ssv/utils/casts"
+)
+
+type Exporter struct {
+	DomainType spectypes.DomainType
+	QBFTStores *ibftstorage.QBFTStores
+}
+
+type ParticipantResponse struct {
+	Role      string `json:"role"`
+	Slot      uint64 `json:"slot"`
+	PublicKey string `json:"public_key"`
+	Message   struct {
+		// We're keeping "Signers" capitalized to avoid breaking existing clients that rely on the current structure
+		Signers []uint64 `json:"Signers"`
+	} `json:"message"`
+}
+
+func (e *Exporter) Decideds(w http.ResponseWriter, r *http.Request) error {
+	var request struct {
+		From    uint64        `json:"from"`
+		To      uint64        `json:"to"`
+		Roles   api.RoleSlice `json:"roles"`
+		PubKeys api.HexSlice  `json:"pubkeys"`
+	}
+	var response struct {
+		Data []*ParticipantResponse `json:"data"`
+	}
+
+	if err := api.Bind(r, &request); err != nil {
+		return api.BadRequestError(err)
+	}
+
+	if request.From > request.To {
+		return api.BadRequestError(fmt.Errorf("'from' must be less than or equal to 'to'"))
+	}
+
+	if len(request.PubKeys) == 0 {
+		return api.BadRequestError(fmt.Errorf("at least one public key is required"))
+	}
+
+	if len(request.Roles) == 0 {
+		return api.BadRequestError(fmt.Errorf("at least one role is required"))
+	}
+
+	response.Data = []*ParticipantResponse{}
+	for _, role := range request.Roles {
+		runnerRole := casts.BeaconRoleToConvertRole(spectypes.BeaconRole(role))
+		roleStorage := e.QBFTStores.Get(runnerRole)
+		if roleStorage == nil {
+			return fmt.Errorf("role storage doesn't exist: %v", role)
+		}
```

Comment:
- maybe its better to check for existence of all roles before this loop. 
-  why not returning `api.BadRequestError` here? 
- maybe we should continue to next role if some of them do exist?

Reply:
I agree with the first two, thanks. I think we don't need to continue if a role doesn't exist because it would indicate that QBFT stores are not setup correctly